docs/docs/server/responses-api.md

```diff
@@ -0,0 +1,528 @@
+
+### Stateful Conversations
+
+In stateful conversations, the server manages conversation history. You reference previous messages using the `previous_response_id` parameter, and the server maintains the context automatically.
+
+By setting `previous_response_id`, all messages in conversation are stored together in the database. This makes conversations easier to analyze.
+
+**Requirements for stateful conversations:**
+1. Must set `store: true | undefined`
+1. Must use the same `user` ID for all requests in the conversation
+1. Maximum of 50 user messages per conversation
+1. Each follow-up must reference the immediately previous response ID
+
+If any of these requirements are violated, the server responds with an error.
+
+```ts
+import { OpenAI } from "openai";
+
+const openai = new OpenAI({ baseURL: "https://knowledge.mongodb.com/api/v1" });
+const userId = "user123";
+
+// First message in conversation
+const firstResponse = await openai.responses.create({
+  model: "mongodb-chat-latest",
+  stream: true,
+  input: "What is MongoDB?",
+  store: true, // Required for stateful conversations
+  user: userId
+});
+
+let previousResponseId: string;
+let conversationId: string;
+
+for await (const event of firstResponse) {
+  switch (event.type) {
+    case "response.completed":
+      previousResponseId = event.response.id;
+      conversationId = event.response.metadata.conversation_id;
+      break;
+  }
+}
+
+// Follow-up message using previous_response_id
+const followUpResponse = await openai.responses.create({
+  model: "mongodb-chat-latest",
+  stream: true,
+  input: "How do I install it?",
+  previous_response_id: previousResponseId, // Links to previous conversation
+  store: true,
+  user: userId // Must be the same user ID
+});
+```
+
+### Stateless Conversations
+
+In stateless conversations, you provide the entire conversation context in each request.
+
+You can use stateless conversations for any storage setting,  `store: true | false | undefined`. 
+
+If you set `store: false`, you **must** use a stateless conversation because the server doesn't maintain conversation history.
+
+```ts
+import { OpenAI } from "openai";
+
+const openai = new OpenAI({ baseURL: "https://knowledge.mongodb.com/api/v1" });
+
+const firstMessage = {
+  type: "message",
+  role: "user",
+  content: "What is MongoDB?"
+}
+// First message
+const firstResponse = await openai.responses.create({
+  model: "mongodb-chat-latest",
+  stream: true,
+  input: [firstMessage],
+  store: false // Does not store conversation on server
+});
+
+
+// Follow-up message with full conversation history
+const followUpResponse = await openai.responses.create({
+  model: "mongodb-chat-latest",
+  stream: true,
+  input: [
+    // Passing first message again because stateless conversation
+    firstMessage,
+    {
+      type: "message",
+      role: "assistant",
+      content: "...some assistant response..."
```

Comment:
nit - this should probably be a reference to `firstResponse` instead of hardcoded string